rpc/src/module/trace.rs

```diff
@@ -31,22 +35,47 @@ impl<CI: ChainIndex + 'static> TraceRpc for TraceRpcImpl<CI> {
     fn trace_transaction(&self, tx: Transaction) -> Result<H256> {
         let tx: CoreTransaction = tx.into();
         let tx_hash = tx.hash().clone();
-        let mut chain_state = self.shared.chain_state().lock();
-        let tx_pool = chain_state.mut_tx_pool();
-        tx_pool.trace_tx(tx.clone());
+        let tx_result = {
+            let chain_state = self.shared.chain_state().lock();
+            let rtx = chain_state.resolve_tx_from_pool(&tx, &chain_state.tx_pool());
+            chain_state.verify_rtx(&rtx, self.shared.consensus().max_block_cycles())
+        };
 
-        let fbb = &mut FlatBufferBuilder::new();
-        let message = RelayMessage::build_transaction(fbb, &tx);
-        fbb.finish(message, None);
+        match tx_result {
+            Ok(cycles) => {
+                let entry = PoolEntry::new(tx.clone(), 0, Some(cycles));
+                self.shared
+                    .chain_state()
+                    .lock()
```

Comment:
lock acquired twice

Reply:
The intention is to release the lock before  broad casting

Reply:
```